.github/workflows/build.yaml

```diff
@@ -0,0 +1,104 @@
+# Copyright 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+---
+name: Build
+
+'on':
+  workflow_call:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build-antora:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: 'antora/package.json'
+          cache: 'npm'
+          cache-dependency-path: 'antora/package-lock.json'
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v4
+        with:
+          go-version-file: antora/go.mod
+          cache: true
+          cache-dependency-path: antora/go.sum
+
+      - name: Generate website
+        run: npm --prefix antora ci && npm --prefix antora run build
+
+      - name: Upload artifact
```

Comment:
I see this workflow can be triggered on pull requests. Will that cause any issues when multiple PRs are uploading artifacts? Not sure if they're properly scoped to the PR. Maybe the `path` below need to have the PR number?

UPDATE: The `preview` workflow seems to indicate that artifacts are properly scoped to a specific run. :tada:

Reply:
Artifacts are property of workflow runs